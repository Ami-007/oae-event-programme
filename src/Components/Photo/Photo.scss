@use "../../sass/variables" as *;
@use "../../sass/typography" as *;

