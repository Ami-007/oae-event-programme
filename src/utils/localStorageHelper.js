export const getDarkTheme = () => {
  //   if (darkTheme == true) {
  //     localStorage.setItem("--background-color", darkTheme);
  //   } else {
  //     localStorage.setItem("theme", "light");
  //   }
  console.log(localStorage.getItem("darkMode"));

  // convert isDarkMode to a Boolean
  // return that boolean

  return true;
};

export const setDarkTheme = (isDarkMode) => {
  localStorage.setItem("darkMode", isDarkMode.toString());
};
