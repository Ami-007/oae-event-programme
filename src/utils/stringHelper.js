export const getFontType = (fontType) => {
  if (typeof fontType === "string") {
    fontType = parseInt(fontType);
  }
  if (fontType === 1) {
    return "modern";
  } else if (fontType === 2) {
    return "classic";
  } else {
    return "modern";
  }
};

export default getFontType;
