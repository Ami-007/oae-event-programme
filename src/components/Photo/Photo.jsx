// src, desciption as props
import React from "react";
import "./Photo.scss";

const Photo = (props) => {
  const { imageSrc, description } = props;

  return (
    <div className="photo">
      <img src={imageSrc} alt="placeholder" className="photo__image" />
      <span className="photo__description">{description}</span>
    </div>
  );
};

export default Photo;
