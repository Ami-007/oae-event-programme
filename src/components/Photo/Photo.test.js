import Photo from "./Photo";
import { customRender } from "../../utils/testUtils";
import testImage from "../../assets/images/image.png";

test("Renders Photo component", () => {
  const { container } = customRender(<Photo />);
  expect(container).toMatchSnapshot();
});

test("Renders imageSrc prop with test image", () => {
  const { container } = customRender(<Photo imageSrc={testImage} />);
  expect(container).toMatchSnapshot();
});

test("Renders description prop with text 'Hello world'", () => {
  const { container } = customRender(<Photo description="Hello world" />);
  expect(container).toMatchSnapshot();
});
