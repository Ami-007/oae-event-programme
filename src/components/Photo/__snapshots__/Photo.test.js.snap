// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Photo component 1`] = `
<div>
  <div
    class="photo"
  >
    <img
      alt="placeholder"
      class="photo__image"
    />
    <p
      class="photo__description"
    />
  </div>
</div>
`;

exports[`Renders description prop with text 'Hello world' 1`] = `
<div>
  <div
    class="photo"
  >
    <img
      alt="placeholder"
      class="photo__image"
    />
    <p
      class="photo__description"
    >
      Hello world
    </p>
  </div>
</div>
`;

exports[`Renders imageSrc prop with test image 1`] = `
<div>
  <div
    class="photo"
  >
    <img
      alt="placeholder"
      class="photo__image"
      src="image.png"
    />
    <p
      class="photo__description"
    />
  </div>
</div>
`;
